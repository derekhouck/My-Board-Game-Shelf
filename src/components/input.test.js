import React from 'react';
import { shallow } from 'enzyme';
import Input from './input';

describe('<Input />', function () {
  it('renders without crashing', () => {
    shallow(<Input meta={{}} input={{}} />);
  });

  it('shows an error message when error exists', function () {
    const wrapper = shallow(<Input meta={{
      touched: true,
      error: 'Test error'
    }} input={{}} />);
    expect(wrapper.exists('.form-error')).toBe(true);
  });

  it('shows a warning message when warning exists', function () {
    const wrapper = shallow(<Input meta={{
      touched: true,
      warning: 'Test warning'
    }} input={{}} />);
    expect(wrapper.exists('.form-warning')).toBe(true);
  });
});
