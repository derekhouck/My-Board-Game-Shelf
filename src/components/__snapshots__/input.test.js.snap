// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    input={Object {}}
    meta={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label />,
        <input />,
      ],
      "className": "form-input undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
            undefined,
          ],
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
          undefined,
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": undefined,
          "type": undefined,
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label />,
          <input />,
        ],
        "className": "form-input undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
            ],
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            undefined,
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": undefined,
            "type": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
